// @cloudflare/realtimekit-ui/dist/esm/rtk-settings-toggle.entry.js
import { r as registerInstance, d as createEvent, h, a as Host } from './index-55463a54.js';
import { e as defaultIconPack, i as useLanguage } from './ui-store-04f890e8.js';
import { S as SyncWithStore } from './index-266a8fa9.js';

const rtkSettingsToggleCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:block}";
const RtkSettingsToggleStyle0 = rtkSettingsToggleCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkSettingsToggle = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.stateUpdate = createEvent(this, "rtkStateUpdate", 7);
        /** Variant */
        this.variant = 'button';
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    toggleSettings() {
        const updatePartial = { activeSettings: true, activeMoreMenu: false };
        this.states = Object.assign(Object.assign({}, this.states), updatePartial);
        this.stateUpdate.emit(updatePartial);
    }
    render() {
        const text = this.t('settings');
        return (h(Host, { key: '921afc41ed4a6019ff882c896855ab455b3edfb7', title: text }, h("rtk-controlbar-button", { key: '88d4194fee24f1b96769e06e3a935d516ad543b0', part: "controlbar-button", size: this.size, iconPack: this.iconPack, onClick: () => this.toggleSettings(), icon: this.iconPack.settings, label: text, variant: this.variant })));
    }
};
__decorate([
    SyncWithStore()
], RtkSettingsToggle.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], RtkSettingsToggle.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkSettingsToggle.prototype, "t", void 0);
RtkSettingsToggle.style = RtkSettingsToggleStyle0;

export { RtkSettingsToggle as rtk_settings_toggle };
