// @cloudflare/realtimekit-ui/dist/esm/rtk-information-tooltip.entry.js
import { r as registerInstance, h, a as Host } from './index-55463a54.js';
import { S as SyncWithStore } from './index-266a8fa9.js';
import { e as defaultIconPack } from './ui-store-04f890e8.js';

const rtkInformationTooltipCss = ":host{margin-left:var(--rtk-space-2, 8px);margin-right:var(--rtk-space-2, 8px);cursor:pointer}.tooltip-container{position:relative;display:flex;flex-direction:row;align-items:center}.tooltip-container rtk-icon{cursor:pointer;color:rgb(var(--rtk-colors-text-600, 255 255 255 / 0.52))}.tooltip-container rtk-icon:hover+.tooltip{display:flex !important}.tooltip{position:absolute;margin-left:var(--rtk-space-2, 8px);display:none !important;border-radius:var(--rtk-border-radius-md, 8px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-700, 44 44 44) / var(--tw-bg-opacity));padding:var(--rtk-space-2, 8px);font-weight:400;color:rgb(var(--rtk-colors-text-600, 255 255 255 / 0.52));z-index:50;display:flex;min-width:var(--rtk-space-60, 240px);flex-direction:column;--tw-shadow:0 4px 6px -1px rgb(0 0 0 / 0.1), 0 2px 4px -2px rgb(0 0 0 / 0.1);--tw-shadow-colored:0 4px 6px -1px var(--tw-shadow-color), 0 2px 4px -2px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);left:14px}";
const RtkInformationTooltipStyle0 = rtkInformationTooltipCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkInformationTooltip = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /** Icon pack */
        this.iconPack = defaultIconPack;
    }
    render() {
        return (h(Host, { key: 'f6ceb562ad29d9bcafe1faaa7f8680cb968509bf' }, h("div", { key: 'f4136f844d6ba57d43e4af5aae03a54a8d439028', class: "tooltip-container" }, h("rtk-icon", { key: 'c6ddee8984ae5f6560484b970bdc10a832be06ec', icon: this.iconPack.info, size: "sm" }), h("div", { key: '78e4eefcd29ee99dcc9fba0cbf015b39b0097348', class: "tooltip" }, h("slot", { key: '1561ac818363c0cbb031a4b4a10d1a56a1dbe379', name: "tootlip-text" })))));
    }
};
__decorate([
    SyncWithStore()
], RtkInformationTooltip.prototype, "iconPack", void 0);
RtkInformationTooltip.style = RtkInformationTooltipStyle0;

export { RtkInformationTooltip as rtk_information_tooltip };
