// @cloudflare/realtimekit-ui/dist/esm/rtk-header.entry.js
import { r as registerInstance, h, a as Host } from './index-55463a54.js';
import { c as createDefaultConfig, e as defaultIconPack, i as useLanguage } from './ui-store-04f890e8.js';
import { R as Render } from './index-3f51790f.js';
import { S as SyncWithStore } from './index-266a8fa9.js';

const rtkHeaderCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:flex;height:var(--rtk-space-12, 48px);align-items:center;--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--rtk-space-4, 16px);padding-right:var(--rtk-space-4, 16px)}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{display:none !important}}@media only screen and (max-height: 480px) and (orientation: landscape){:host{display:none !important}}";
const RtkHeaderStyle0 = rtkHeaderCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkHeader = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = createDefaultConfig();
        /** Icon Pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (h(Host, { key: 'e88e21ff9c7d6499eb935b4ed487eea19a5da28b' }, !this.disableRender && h(Render, { key: '1e6641216a381c9d07152f61eb2a3c0634871bde', element: "rtk-header", defaults: defaults, onlyChildren: true }), h("slot", { key: '0cc84f35845f14f426c2b68c9abe914e9654f2db' })));
    }
};
__decorate([
    SyncWithStore()
], RtkHeader.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], RtkHeader.prototype, "config", void 0);
__decorate([
    SyncWithStore()
], RtkHeader.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], RtkHeader.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkHeader.prototype, "t", void 0);
RtkHeader.style = RtkHeaderStyle0;

export { RtkHeader as rtk_header };
