// @cloudflare/realtimekit-ui/dist/esm/rtk-debugger-toggle.entry.js
import { r as registerInstance, d as createEvent, h, a as Host } from './index-55463a54.js';
import { i as useLanguage, e as defaultIconPack } from './ui-store-04f890e8.js';
import { S as SyncWithStore } from './index-266a8fa9.js';

const rtkDebuggerToggleCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:block}";
const RtkDebuggerToggleStyle0 = rtkDebuggerToggleCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkDebuggerToggle = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.stateUpdate = createEvent(this, "rtkStateUpdate", 7);
        /** Variant */
        this.variant = 'button';
        /** Language */
        this.t = useLanguage();
        /** Icon pack */
        this.iconPack = defaultIconPack;
    }
    toggleDebugger() {
        var _a;
        this.stateUpdate.emit({
            activeDebugger: !((_a = this.states) === null || _a === void 0 ? void 0 : _a.activeDebugger),
            activeMoreMenu: false,
        });
    }
    render() {
        return (h(Host, { key: '3a400514eae52d10e947446e952e89f8ab539ba9', title: this.t('Troubleshooting') }, h("rtk-controlbar-button", { key: 'ad92157ea3d66898cfbc1bd65d3ec72720cf5e16', size: this.size, iconPack: this.iconPack, onClick: () => this.toggleDebugger(), icon: this.iconPack.debug, label: this.t('Troubleshooting'), variant: this.variant })));
    }
};
__decorate([
    SyncWithStore()
], RtkDebuggerToggle.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], RtkDebuggerToggle.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], RtkDebuggerToggle.prototype, "t", void 0);
__decorate([
    SyncWithStore()
], RtkDebuggerToggle.prototype, "iconPack", void 0);
RtkDebuggerToggle.style = RtkDebuggerToggleStyle0;

export { RtkDebuggerToggle as rtk_debugger_toggle };
