// @cloudflare/realtimekit-ui/dist/esm/rtk-image-viewer.entry.js
import { r as registerInstance, d as createEvent, h, a as Host } from './index-55463a54.js';
import { i as useLanguage, e as defaultIconPack } from './ui-store-04f890e8.js';
import { d as downloadFile } from './file-6114f1a9.js';
import { S as SyncWithStore } from './index-266a8fa9.js';
import { f as formatName, a as shorten } from './string-068ce30a.js';

const rtkImageViewerCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}.scrollbar{scrollbar-width:thin;scrollbar-color:var(--rtk-scrollbar-color, rgb(var(--rtk-colors-background-600, 60 60 60)))\n    var(--rtk-scrollbar-background, transparent)}.scrollbar::-webkit-scrollbar{height:var(--rtk-space-1\\.5, 6px);width:var(--rtk-space-1\\.5, 6px);border-radius:9999px;background-color:var(--rtk-scrollbar-background, transparent)}.scrollbar::-webkit-scrollbar-thumb{border-radius:9999px;background-color:var(--rtk-scrollbar-color, rgb(var(--rtk-colors-background-600, 60 60 60)))}:host{width:1140px;max-width:100%;box-sizing:border-box;display:flex;flex-direction:column;padding:var(--rtk-space-6, 24px);padding-top:var(--rtk-space-5, 20px);overflow-y:auto;color:rgb(var(--rtk-colors-text-1000, 255 255 255));z-index:40;border-radius:var(--rtk-border-radius-md, 8px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-900, 26 26 26) / var(--tw-bg-opacity))}.displayName{font-weight:700}.image-ctr{margin-top:var(--rtk-space-2, 8px);box-sizing:border-box;display:flex;justify-content:center;overflow:hidden}.actions{display:flex;align-items:center;justify-content:flex-end;gap:var(--rtk-space-2, 8px)}img{box-sizing:border-box;display:block;max-height:100%;max-width:100%;-o-object-fit:contain;object-fit:contain}.header{display:flex;align-items:center;justify-content:space-between;padding-bottom:var(--rtk-space-4, 16px)}.shared-by-user{overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:1}:host([size='sm']) .header{flex-direction:column}:host([size='sm']) .header .actions{margin-top:var(--rtk-space-4, 16px)}";
const RtkImageViewerStyle0 = rtkImageViewerCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkImageViewer = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.close = createEvent(this, "close", 7);
        /** Language */
        this.t = useLanguage();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        this.keypressListener = (e) => {
            if (e.key === 'Escape') {
                this.close.emit();
            }
        };
        this.handleOutsideClick = () => this.close.emit();
    }
    connectedCallback() {
        document.addEventListener('keydown', this.keypressListener);
        document.addEventListener('click', this.handleOutsideClick);
    }
    disconnectedCallback() {
        document.removeEventListener('keydown', this.keypressListener);
        document.removeEventListener('click', this.handleOutsideClick);
    }
    render() {
        return (h(Host, { key: '6f8bcc73b18fd868c085d6447ced2ecb24e42325', class: "scrollbar", onClick: (e) => e.stopPropagation() }, h("div", { key: '11f7ad8c7e89442c0d257c57144a5408d90a178a', class: "header" }, h("div", { key: 'e9d10589aa9ceb961a34e55f74238a17632c4baf', class: "shared-by-user" }, this.t('chat.img.shared_by'), ' ', h("span", { key: '3d37f78f6f9c7a92a91bdc18086e24d2818e9ed8', class: "displayName" }, formatName(shorten(this.image.displayName)))), h("div", { key: 'c551cd3d122d1be1076963de15b3f1d76520151c', class: "actions" }, h("rtk-button", { key: '5433bb1fb5e6dac6925d3116d4836b00cb4c69e1', onClick: () => downloadFile(this.image.link, { fallbackName: 'image' }) }, h("rtk-icon", { key: '93ed055d68686903014fd5ebaee3330643a91859', icon: this.iconPack.download, slot: "start" }), "Download"), h("rtk-button", { key: 'e77e1d1cbd4daaa09f133cf4ca54efcea74fd7b2', kind: "icon", variant: "secondary", onClick: () => this.close.emit() }, h("rtk-icon", { key: '8884eb7345fd6903e00c1c7f86c51abe499066e9', icon: this.iconPack.dismiss })))), h("div", { key: 'b223cb91a00ce7967efe5ebcd1f0c680567ba098', class: "image-ctr" }, h("img", { key: 'b4ffa73bd05908314cb164302e51621a64fa074e', src: this.image.link }))));
    }
};
__decorate([
    SyncWithStore()
], RtkImageViewer.prototype, "t", void 0);
__decorate([
    SyncWithStore()
], RtkImageViewer.prototype, "iconPack", void 0);
RtkImageViewer.style = RtkImageViewerStyle0;

export { RtkImageViewer as rtk_image_viewer };
