// @cloudflare/realtimekit-ui/dist/cjs/rtk-controlbar.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-2e5758b6.js');
const uiStore = require('./ui-store-9d5b8dc5.js');
const index$2 = require('./index-6f8c1928.js');
const index = require('./index-2c1fa663.js');

const rtkControlbarCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{box-sizing:border-box;display:flex;align-items:center;gap:var(--rtk-space-0\\.5, 2px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--rtk-space-4, 16px);padding-right:var(--rtk-space-4, 16px);position:relative;z-index:10}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{padding-top:var(--rtk-space-0, 0px) !important}}";
const RtkControlbarStyle0 = rtkControlbarCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkControlbar = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = uiStore.createDefaultConfig();
        /** Icon Pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (index$1.h(index$1.Host, { key: 'be00ae26f16ab4381eddade1f80172789a2833b2' }, !this.disableRender && (index$1.h(index$2.Render, { key: 'd7ed714d51dfcb94195864d462c1b9598eea8ca7', element: "rtk-controlbar", defaults: defaults, onlyChildren: true })), index$1.h("slot", { key: '1601a47d06eab5c79762abdb45d3edb4b0a9f6b6' })));
    }
};
__decorate([
    index.SyncWithStore()
], RtkControlbar.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], RtkControlbar.prototype, "config", void 0);
__decorate([
    index.SyncWithStore()
], RtkControlbar.prototype, "states", void 0);
__decorate([
    index.SyncWithStore()
], RtkControlbar.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], RtkControlbar.prototype, "t", void 0);
RtkControlbar.style = RtkControlbarStyle0;

exports.rtk_controlbar = RtkControlbar;
