// @cloudflare/realtimekit-ui/dist/cjs/rtk-waiting-screen.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-2e5758b6.js');
const uiStore = require('./ui-store-9d5b8dc5.js');
const index = require('./index-2c1fa663.js');

const rtkWaitingScreenCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{height:100%;width:100%;display:flex;flex-direction:column;align-items:center;justify-content:center}.centered{display:flex;flex-direction:column;align-items:center}rtk-logo{margin-bottom:var(--rtk-space-8, 32px);height:var(--rtk-space-12, 48px)}p{font-size:16px;border-radius:var(--rtk-border-radius-lg, 12px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-800, 30 30 30) / var(--tw-bg-opacity));padding-left:var(--rtk-space-8, 32px);padding-right:var(--rtk-space-8, 32px);padding-top:var(--rtk-space-4, 16px);padding-bottom:var(--rtk-space-4, 16px);color:rgb(var(--rtk-colors-text-1000, 255 255 255))}";
const RtkWaitingScreenStyle0 = rtkWaitingScreenCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkWaitingScreen = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Config */
        this.config = uiStore.createDefaultConfig();
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        return (index$1.h(index$1.Host, { key: 'ac7096bd5df59387dcef15bbd4ef8955a4a42298' }, index$1.h("slot", { key: '6dd0f5b9555c956c1371311ef3c4ac8f661e9523' }, index$1.h("div", { key: '07925b636545e1196b10512cd504516df2cfbbf4', class: "centered", part: "content" }, index$1.h("rtk-logo", { key: '9b3fcf182d7c68aaf525a8170dbab9e59fc40404', meeting: this.meeting, config: this.config, part: "logo", t: this.t }), index$1.h("p", { key: '763a6403bc98a16f020101090bbed250041ae7cd' }, this.t('waitlist.body_text'))))));
    }
};
__decorate([
    index.SyncWithStore()
], RtkWaitingScreen.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], RtkWaitingScreen.prototype, "config", void 0);
__decorate([
    index.SyncWithStore()
], RtkWaitingScreen.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], RtkWaitingScreen.prototype, "t", void 0);
RtkWaitingScreen.style = RtkWaitingScreenStyle0;

exports.rtk_waiting_screen = RtkWaitingScreen;
