// @cloudflare/realtimekit-ui/dist/cjs/rtk-file-dropzone.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-2e5758b6.js');
const uiStore = require('./ui-store-9d5b8dc5.js');
const index = require('./index-2c1fa663.js');

const rtkFileDropzoneCss = "#dropzone{position:absolute;top:var(--rtk-space-0, 0px);right:var(--rtk-space-0, 0px);bottom:var(--rtk-space-0, 0px);left:var(--rtk-space-0, 0px);z-index:10;display:none;flex-direction:column;align-items:center;justify-content:center;--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-800, 30 30 30) / var(--tw-bg-opacity));color:rgb(var(--rtk-colors-text-700, 255 255 255 / 0.64))}#dropzone.active{display:flex;animation:0.2s slide-up ease-in}";
const RtkFileDropzoneStyle0 = rtkFileDropzoneCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkFileDropzone = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        this.onDropCallback = index$1.createEvent(this, "dropCallback", 7);
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        this.dropzoneActivated = false;
    }
    connectedCallback() {
        if (!this.hostEl)
            throw new Error('hostEl prop is required');
        this.hostEl.addEventListener('dragover', (e) => {
            e.preventDefault();
            this.dropzoneActivated = true;
        });
        this.hostEl.addEventListener('dragleave', () => {
            this.dropzoneActivated = false;
        });
        this.hostEl.addEventListener('drop', (e) => {
            e.preventDefault();
            this.dropzoneActivated = false;
            this.onDropCallback.emit(e);
        });
    }
    render() {
        return (index$1.h(index$1.Host, { key: '75d3181f8ae69f463d10eff1befe75cf5856eae9' }, index$1.h("div", { key: '385ec4116e8b1748219d3c6bb803ba49dae07c04', id: "dropzone", class: { active: this.dropzoneActivated }, part: "dropzone" }, index$1.h("rtk-icon", { key: 'fcdb498c515168f9b38e6a8e253484ed2ab48fe9', icon: this.iconPack.attach }), index$1.h("p", { key: '230cbb525a29675e43aa36c8e4d4711c75cedff0' }, this.t('chat.send_attachment')))));
    }
};
__decorate([
    index.SyncWithStore()
], RtkFileDropzone.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], RtkFileDropzone.prototype, "t", void 0);
RtkFileDropzone.style = RtkFileDropzoneStyle0;

exports.rtk_file_dropzone = RtkFileDropzone;
