// @cloudflare/realtimekit-ui/dist/cjs/rtk-file-message_3.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-2e5758b6.js');
const ChatHead = require('./ChatHead-777e6a2d.js');
const string = require('./string-9bce3253.js');
const uiStore = require('./ui-store-9d5b8dc5.js');
const file = require('./file-0b9afa34.js');
const index = require('./index-2c1fa663.js');
const TextMessage = require('./TextMessage-9a4bcde2.js');
require('./date-61ffcd0a.js');
require('./chat-02670cc5.js');

var __decorate$2 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkFileMessage = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Date object of now, to calculate distance between dates */
        this.now = new Date();
        /** Whether the message is continued by same user */
        this.isContinued = false;
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        /** show message in bubble */
        this.showBubble = false;
    }
    render() {
        const link = string.sanitizeLink(this.message.link);
        return (index$1.h(index$1.Host, { key: '76c8da9d878e7c1fb3b8ffa315237cd913338276' }, !this.isContinued && (index$1.h(ChatHead.ChatHead, { key: 'efd7339f8335d7da577f61caa61642fe48f5a412', name: this.message.displayName, time: this.message.time, now: this.now })), index$1.h("div", { key: '53947f39384592b44f42199a3c8eb2f4d90adafd', class: {
                body: true,
                bubble: this.showBubble,
            }, part: "body" }, index$1.h("div", { key: '1a09d5b047cfa88912e8fada39a5cf8b0e138736', class: "file" }, index$1.h("div", { key: 'fc6a30c8ddb0f8ac895de7ffad4bc4c455a59dfa', class: "file-data" }, index$1.h("div", { key: '4ad8fa217942ca3f61faea3ed4de9fd0f158bbb1', class: "name" }, this.message.name), index$1.h("div", { key: 'd4adc2dd11b6be4ce508ac50eea67e8d500f1929', class: "file-data-split" }, index$1.h("div", { key: 'cca022a6825fbafa4c257ac9f32f4d25f2eece2b', class: "ext" }, file.getExtension(this.message.name)), index$1.h("span", { key: '62917b4cb5371045f8fcee541c2cc872038a90f9', class: "divider" }), index$1.h("div", { key: '9cae7724db8096dfa64f2f2e70915fddd0b1fb43', class: "size" }, file.getFileSize(this.message.size)))), index$1.h("rtk-button", { key: 'd2645e34ed2f984fa0403820c47329fa82ae4d58', variant: "secondary", kind: "icon", onClick: () => file.downloadFile(link, { name: this.message.name, fallbackName: 'file' }), part: "button" }, index$1.h("rtk-icon", { key: '5650ea4473efec76a2e8942fa32be671ac143fb7', icon: this.iconPack.download }))))));
    }
};
__decorate$2([
    index.SyncWithStore()
], RtkFileMessage.prototype, "iconPack", void 0);
__decorate$2([
    index.SyncWithStore()
], RtkFileMessage.prototype, "t", void 0);

const rtkImageMessageCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}.image-spinner{cursor:wait}.image-errored{cursor:not-allowed}";
const RtkImageMessageStyle0 = rtkImageMessageCss;

var __decorate$1 = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkImageMessage = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        this.stateUpdate = index$1.createEvent(this, "rtkStateUpdate", 7);
        /** Date object of now, to calculate distance between dates */
        this.now = new Date();
        /** Whether the message is continued by same user */
        this.isContinued = false;
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        /** show message in bubble */
        this.showBubble = false;
        this.status = 'loading';
    }
    render() {
        return (index$1.h(index$1.Host, { key: '0fd97958347fefb4a1666e1d676726d13220e5b3' }, !this.isContinued && (index$1.h(ChatHead.ChatHead, { key: '8a629fe3782724a0c82c57850ee36ede941d83d7', name: this.message.displayName, time: this.message.time, now: this.now })), index$1.h("div", { key: '8e132ad25a04ddb4abb167b6b0e9863670baefac', class: {
                body: true,
                bubble: this.showBubble,
            }, part: "body" }, index$1.h("div", { key: '19cf2331068bc84218b2b7153c49ff1af2dedca8', class: { image: true, loaded: this.status === 'loaded' } }, index$1.h("img", { key: '86e08153c32cb6f78d8477fc1abdb5b8f4cd0ed0', src: string.sanitizeLink(this.message.link), onLoad: () => {
                this.status = 'loaded';
            }, onError: () => {
                this.status = 'errored';
            }, onClick: () => {
                if (this.status === 'loaded') {
                    this.stateUpdate.emit({ image: this.message });
                }
            } }), this.status === 'loading' && (index$1.h("div", { key: '367b76997445b7df0226aaa063b884c764a6ba63', class: "image-spinner", title: this.t('chat.img.loading'), "aria-label": this.t('chat.img.loading') }, index$1.h("rtk-spinner", { key: '704b81940f6f49cb79d2a5976a2714039358e007', iconPack: this.iconPack }))), this.status === 'errored' && (index$1.h("div", { key: 'eb9a5b3593051905c3b4b345c810b318da1cb7cf', class: "image-errored", title: this.t('chat.error.img_not_found'), "aria-label": this.t('chat.error.img_not_found') }, index$1.h("rtk-icon", { key: '73dfa63e1a253eac3dc75e598341335f63abb855', icon: this.iconPack.image_off }))), this.status === 'loaded' && (index$1.h("div", { key: '48444ac8542f8e3b5ed81c93c1b1b1f0b2cac39b', class: "actions" }, index$1.h("rtk-button", { key: '2982aed5152442362a22f873cc7070b60462f7e4', class: "action", variant: "secondary", kind: "icon", onClick: () => {
                this.stateUpdate.emit({ image: this.message });
            } }, index$1.h("rtk-icon", { key: 'c8dbeac2a347245f21eef52f0c42772e48741f93', icon: this.iconPack.full_screen_maximize })), index$1.h("rtk-button", { key: 'cfc3845ea0ae2373bb5a92d897d926797713cbf0', class: "action", variant: "secondary", kind: "icon", onClick: () => file.downloadFile(this.message.link, { fallbackName: 'image' }) }, index$1.h("rtk-icon", { key: '56a6109c8ba88160ef75f8c13923ab12a4764e0e', icon: this.iconPack.download }))))))));
    }
};
__decorate$1([
    index.SyncWithStore()
], RtkImageMessage.prototype, "iconPack", void 0);
__decorate$1([
    index.SyncWithStore()
], RtkImageMessage.prototype, "t", void 0);
RtkImageMessage.style = RtkImageMessageStyle0;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkTextMessage = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Date object of now, to calculate distance between dates */
        this.now = new Date();
        /** Whether the message is continued by same user */
        this.isContinued = false;
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        /** show message in bubble */
        this.showBubble = false;
    }
    render() {
        return (index$1.h(index$1.Host, { key: 'a34dd007a87a134202e153f3d886102bbe6a685c' }, !this.isContinued && (index$1.h(ChatHead.ChatHead, { key: '8505cee0d359b7449776ece2bbd7455811217a41', name: this.message.displayName, time: this.message.time, now: this.now })), index$1.h("div", { key: '00b2baca454b62e4984e82790580bac116a046a0', class: {
                body: true,
                bubble: this.showBubble,
            }, part: "body" }, index$1.h("div", { key: 'ecc8de946011c9836cebfd626a27cb23f111d0cc', class: { text: true, emoji: string.hasOnlyEmojis(this.message.message) } }, index$1.h(TextMessage.TextMessageView, { key: '1efad576e6110986cb2abac7ff2c42b492c995aa', message: this.message.message })))));
    }
};
__decorate([
    index.SyncWithStore()
], RtkTextMessage.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], RtkTextMessage.prototype, "t", void 0);

exports.rtk_file_message = RtkFileMessage;
exports.rtk_image_message = RtkImageMessage;
exports.rtk_text_message = RtkTextMessage;
