// @cloudflare/realtimekit-ui/dist/cjs/rtk-header.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-2e5758b6.js');
const uiStore = require('./ui-store-9d5b8dc5.js');
const index$2 = require('./index-6f8c1928.js');
const index = require('./index-2c1fa663.js');

const rtkHeaderCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:flex;height:var(--rtk-space-12, 48px);align-items:center;--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--rtk-space-4, 16px);padding-right:var(--rtk-space-4, 16px)}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{display:none !important}}@media only screen and (max-height: 480px) and (orientation: landscape){:host{display:none !important}}";
const RtkHeaderStyle0 = rtkHeaderCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkHeader = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = uiStore.createDefaultConfig();
        /** Icon Pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (index$1.h(index$1.Host, { key: 'e88e21ff9c7d6499eb935b4ed487eea19a5da28b' }, !this.disableRender && index$1.h(index$2.Render, { key: '1e6641216a381c9d07152f61eb2a3c0634871bde', element: "rtk-header", defaults: defaults, onlyChildren: true }), index$1.h("slot", { key: '0cc84f35845f14f426c2b68c9abe914e9654f2db' })));
    }
};
__decorate([
    index.SyncWithStore()
], RtkHeader.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], RtkHeader.prototype, "config", void 0);
__decorate([
    index.SyncWithStore()
], RtkHeader.prototype, "states", void 0);
__decorate([
    index.SyncWithStore()
], RtkHeader.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], RtkHeader.prototype, "t", void 0);
RtkHeader.style = RtkHeaderStyle0;

exports.rtk_header = RtkHeader;
