// @cloudflare/realtimekit-ui/dist/cjs/rtk-chat-search-results.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-2e5758b6.js');
const uiStore = require('./ui-store-9d5b8dc5.js');
const index = require('./index-2c1fa663.js');

const rtkChatSearchResultsCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:flex;height:100%;flex-direction:column;position:relative;--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-900, 26 26 26) / var(--tw-bg-opacity))}";
const RtkChatSearchResultsStyle0 = rtkChatSearchResultsCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkChatSearchResults = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        this.pageSize = 50;
        this.searchMessages = async (timestamp, size, reversed) => {
            return this.meeting.chat.searchMessages(this.query, {
                channelId: this.channelId,
                timestamp,
                size,
                reversed,
            });
        };
        this.nodeRenderer = (messages) => {
            return messages.map((message) => (index$1.h("rtk-chat-message", { key: message.id, message: message, disableControls: true })));
        };
    }
    render() {
        return (index$1.h(index$1.Host, { key: 'dcde089188305612e2f931128f0eedd3c06724f9' }, index$1.h("rtk-paginated-list", { key: '0e73fa0aa2e8696b0aa886ca80e6e22ffb725cf5', pageSize: this.pageSize, pagesAllowed: 3, fetchData: this.searchMessages, createNodes: this.nodeRenderer })));
    }
};
__decorate([
    index.SyncWithStore()
], RtkChatSearchResults.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], RtkChatSearchResults.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], RtkChatSearchResults.prototype, "t", void 0);
RtkChatSearchResults.style = RtkChatSearchResultsStyle0;

exports.rtk_chat_search_results = RtkChatSearchResults;
