// @cloudflare/realtimekit-react-ui/dist/es/components/stencil-generated/index.js
/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';
import { defineCustomElements } from '@cloudflare/realtimekit-ui/loader';
defineCustomElements();
export const RtkAi = /*@__PURE__*/ createReactComponent('rtk-ai');
export const RtkAiToggle = /*@__PURE__*/ createReactComponent('rtk-ai-toggle');
export const RtkAiTranscriptions = /*@__PURE__*/ createReactComponent('rtk-ai-transcriptions');
export const RtkAudioGrid = /*@__PURE__*/ createReactComponent('rtk-audio-grid');
export const RtkAudioTile = /*@__PURE__*/ createReactComponent('rtk-audio-tile');
export const RtkAudioVisualizer = /*@__PURE__*/ createReactComponent('rtk-audio-visualizer');
export const RtkAvatar = /*@__PURE__*/ createReactComponent('rtk-avatar');
export const RtkBreakoutRoomManager = /*@__PURE__*/ createReactComponent('rtk-breakout-room-manager');
export const RtkBreakoutRoomParticipants = /*@__PURE__*/ createReactComponent('rtk-breakout-room-participants');
export const RtkBreakoutRoomsManager = /*@__PURE__*/ createReactComponent('rtk-breakout-rooms-manager');
export const RtkBreakoutRoomsToggle = /*@__PURE__*/ createReactComponent('rtk-breakout-rooms-toggle');
export const RtkBroadcastMessageModal = /*@__PURE__*/ createReactComponent('rtk-broadcast-message-modal');
export const RtkButton = /*@__PURE__*/ createReactComponent('rtk-button');
export const RtkCameraSelector = /*@__PURE__*/ createReactComponent('rtk-camera-selector');
export const RtkCameraToggle = /*@__PURE__*/ createReactComponent('rtk-camera-toggle');
export const RtkCaptionToggle = /*@__PURE__*/ createReactComponent('rtk-caption-toggle');
export const RtkChat = /*@__PURE__*/ createReactComponent('rtk-chat');
export const RtkChatComposerUi = /*@__PURE__*/ createReactComponent('rtk-chat-composer-ui');
export const RtkChatComposerView = /*@__PURE__*/ createReactComponent('rtk-chat-composer-view');
export const RtkChatHeader = /*@__PURE__*/ createReactComponent('rtk-chat-header');
export const RtkChatMessage = /*@__PURE__*/ createReactComponent('rtk-chat-message');
export const RtkChatMessagesUi = /*@__PURE__*/ createReactComponent('rtk-chat-messages-ui');
export const RtkChatMessagesUiPaginated = /*@__PURE__*/ createReactComponent('rtk-chat-messages-ui-paginated');
export const RtkChatSearchResults = /*@__PURE__*/ createReactComponent('rtk-chat-search-results');
export const RtkChatSelector = /*@__PURE__*/ createReactComponent('rtk-chat-selector');
export const RtkChatSelectorUi = /*@__PURE__*/ createReactComponent('rtk-chat-selector-ui');
export const RtkChatToggle = /*@__PURE__*/ createReactComponent('rtk-chat-toggle');
export const RtkClock = /*@__PURE__*/ createReactComponent('rtk-clock');
export const RtkConfirmationModal = /*@__PURE__*/ createReactComponent('rtk-confirmation-modal');
export const RtkControlbar = /*@__PURE__*/ createReactComponent('rtk-controlbar');
export const RtkControlbarButton = /*@__PURE__*/ createReactComponent('rtk-controlbar-button');
export const RtkCounter = /*@__PURE__*/ createReactComponent('rtk-counter');
export const RtkDebugger = /*@__PURE__*/ createReactComponent('rtk-debugger');
export const RtkDebuggerAudio = /*@__PURE__*/ createReactComponent('rtk-debugger-audio');
export const RtkDebuggerScreenshare = /*@__PURE__*/ createReactComponent('rtk-debugger-screenshare');
export const RtkDebuggerSystem = /*@__PURE__*/ createReactComponent('rtk-debugger-system');
export const RtkDebuggerToggle = /*@__PURE__*/ createReactComponent('rtk-debugger-toggle');
export const RtkDebuggerVideo = /*@__PURE__*/ createReactComponent('rtk-debugger-video');
export const RtkDialog = /*@__PURE__*/ createReactComponent('rtk-dialog');
export const RtkDialogManager = /*@__PURE__*/ createReactComponent('rtk-dialog-manager');
export const RtkDraftAttachmentView = /*@__PURE__*/ createReactComponent('rtk-draft-attachment-view');
export const RtkEmojiPicker = /*@__PURE__*/ createReactComponent('rtk-emoji-picker');
export const RtkEmojiPickerButton = /*@__PURE__*/ createReactComponent('rtk-emoji-picker-button');
export const RtkEndedScreen = /*@__PURE__*/ createReactComponent('rtk-ended-screen');
export const RtkFileDropzone = /*@__PURE__*/ createReactComponent('rtk-file-dropzone');
export const RtkFileMessage = /*@__PURE__*/ createReactComponent('rtk-file-message');
export const RtkFileMessageView = /*@__PURE__*/ createReactComponent('rtk-file-message-view');
export const RtkFilePickerButton = /*@__PURE__*/ createReactComponent('rtk-file-picker-button');
export const RtkFullscreenToggle = /*@__PURE__*/ createReactComponent('rtk-fullscreen-toggle');
export const RtkGrid = /*@__PURE__*/ createReactComponent('rtk-grid');
export const RtkGridPagination = /*@__PURE__*/ createReactComponent('rtk-grid-pagination');
export const RtkHeader = /*@__PURE__*/ createReactComponent('rtk-header');
export const RtkIcon = /*@__PURE__*/ createReactComponent('rtk-icon');
export const RtkIdleScreen = /*@__PURE__*/ createReactComponent('rtk-idle-screen');
export const RtkImageMessage = /*@__PURE__*/ createReactComponent('rtk-image-message');
export const RtkImageMessageView = /*@__PURE__*/ createReactComponent('rtk-image-message-view');
export const RtkImageViewer = /*@__PURE__*/ createReactComponent('rtk-image-viewer');
export const RtkInformationTooltip = /*@__PURE__*/ createReactComponent('rtk-information-tooltip');
export const RtkJoinStage = /*@__PURE__*/ createReactComponent('rtk-join-stage');
export const RtkLeaveButton = /*@__PURE__*/ createReactComponent('rtk-leave-button');
export const RtkLeaveMeeting = /*@__PURE__*/ createReactComponent('rtk-leave-meeting');
export const RtkLivestreamIndicator = /*@__PURE__*/ createReactComponent('rtk-livestream-indicator');
export const RtkLivestreamPlayer = /*@__PURE__*/ createReactComponent('rtk-livestream-player');
export const RtkLivestreamToggle = /*@__PURE__*/ createReactComponent('rtk-livestream-toggle');
export const RtkLogo = /*@__PURE__*/ createReactComponent('rtk-logo');
export const RtkMarkdownView = /*@__PURE__*/ createReactComponent('rtk-markdown-view');
export const RtkMeeting = /*@__PURE__*/ createReactComponent('rtk-meeting');
export const RtkMeetingTitle = /*@__PURE__*/ createReactComponent('rtk-meeting-title');
export const RtkMenu = /*@__PURE__*/ createReactComponent('rtk-menu');
export const RtkMenuItem = /*@__PURE__*/ createReactComponent('rtk-menu-item');
export const RtkMenuList = /*@__PURE__*/ createReactComponent('rtk-menu-list');
export const RtkMessageListView = /*@__PURE__*/ createReactComponent('rtk-message-list-view');
export const RtkMessageView = /*@__PURE__*/ createReactComponent('rtk-message-view');
export const RtkMicToggle = /*@__PURE__*/ createReactComponent('rtk-mic-toggle');
export const RtkMicrophoneSelector = /*@__PURE__*/ createReactComponent('rtk-microphone-selector');
export const RtkMixedGrid = /*@__PURE__*/ createReactComponent('rtk-mixed-grid');
export const RtkMoreToggle = /*@__PURE__*/ createReactComponent('rtk-more-toggle');
export const RtkMuteAllButton = /*@__PURE__*/ createReactComponent('rtk-mute-all-button');
export const RtkMuteAllConfirmation = /*@__PURE__*/ createReactComponent('rtk-mute-all-confirmation');
export const RtkNameTag = /*@__PURE__*/ createReactComponent('rtk-name-tag');
export const RtkNetworkIndicator = /*@__PURE__*/ createReactComponent('rtk-network-indicator');
export const RtkNotification = /*@__PURE__*/ createReactComponent('rtk-notification');
export const RtkNotifications = /*@__PURE__*/ createReactComponent('rtk-notifications');
export const RtkOverlayModal = /*@__PURE__*/ createReactComponent('rtk-overlay-modal');
export const RtkPaginatedList = /*@__PURE__*/ createReactComponent('rtk-paginated-list');
export const RtkParticipant = /*@__PURE__*/ createReactComponent('rtk-participant');
export const RtkParticipantCount = /*@__PURE__*/ createReactComponent('rtk-participant-count');
export const RtkParticipantSetup = /*@__PURE__*/ createReactComponent('rtk-participant-setup');
export const RtkParticipantTile = /*@__PURE__*/ createReactComponent('rtk-participant-tile');
export const RtkParticipants = /*@__PURE__*/ createReactComponent('rtk-participants');
export const RtkParticipantsAudio = /*@__PURE__*/ createReactComponent('rtk-participants-audio');
export const RtkParticipantsStageList = /*@__PURE__*/ createReactComponent('rtk-participants-stage-list');
export const RtkParticipantsStageQueue = /*@__PURE__*/ createReactComponent('rtk-participants-stage-queue');
export const RtkParticipantsToggle = /*@__PURE__*/ createReactComponent('rtk-participants-toggle');
export const RtkParticipantsViewerList = /*@__PURE__*/ createReactComponent('rtk-participants-viewer-list');
export const RtkParticipantsWaitingList = /*@__PURE__*/ createReactComponent('rtk-participants-waiting-list');
export const RtkPermissionsMessage = /*@__PURE__*/ createReactComponent('rtk-permissions-message');
export const RtkPinnedMessageSelector = /*@__PURE__*/ createReactComponent('rtk-pinned-message-selector');
export const RtkPipToggle = /*@__PURE__*/ createReactComponent('rtk-pip-toggle');
export const RtkPluginMain = /*@__PURE__*/ createReactComponent('rtk-plugin-main');
export const RtkPlugins = /*@__PURE__*/ createReactComponent('rtk-plugins');
export const RtkPluginsToggle = /*@__PURE__*/ createReactComponent('rtk-plugins-toggle');
export const RtkPoll = /*@__PURE__*/ createReactComponent('rtk-poll');
export const RtkPollForm = /*@__PURE__*/ createReactComponent('rtk-poll-form');
export const RtkPolls = /*@__PURE__*/ createReactComponent('rtk-polls');
export const RtkPollsToggle = /*@__PURE__*/ createReactComponent('rtk-polls-toggle');
export const RtkRecordingIndicator = /*@__PURE__*/ createReactComponent('rtk-recording-indicator');
export const RtkRecordingToggle = /*@__PURE__*/ createReactComponent('rtk-recording-toggle');
export const RtkScreenShareToggle = /*@__PURE__*/ createReactComponent('rtk-screen-share-toggle');
export const RtkScreenshareView = /*@__PURE__*/ createReactComponent('rtk-screenshare-view');
export const RtkSettings = /*@__PURE__*/ createReactComponent('rtk-settings');
export const RtkSettingsAudio = /*@__PURE__*/ createReactComponent('rtk-settings-audio');
export const RtkSettingsToggle = /*@__PURE__*/ createReactComponent('rtk-settings-toggle');
export const RtkSettingsVideo = /*@__PURE__*/ createReactComponent('rtk-settings-video');
export const RtkSetupScreen = /*@__PURE__*/ createReactComponent('rtk-setup-screen');
export const RtkSidebar = /*@__PURE__*/ createReactComponent('rtk-sidebar');
export const RtkSidebarUi = /*@__PURE__*/ createReactComponent('rtk-sidebar-ui');
export const RtkSimpleGrid = /*@__PURE__*/ createReactComponent('rtk-simple-grid');
export const RtkSpeakerSelector = /*@__PURE__*/ createReactComponent('rtk-speaker-selector');
export const RtkSpinner = /*@__PURE__*/ createReactComponent('rtk-spinner');
export const RtkSpotlightGrid = /*@__PURE__*/ createReactComponent('rtk-spotlight-grid');
export const RtkSpotlightIndicator = /*@__PURE__*/ createReactComponent('rtk-spotlight-indicator');
export const RtkStage = /*@__PURE__*/ createReactComponent('rtk-stage');
export const RtkStageToggle = /*@__PURE__*/ createReactComponent('rtk-stage-toggle');
export const RtkSwitch = /*@__PURE__*/ createReactComponent('rtk-switch');
export const RtkTabBar = /*@__PURE__*/ createReactComponent('rtk-tab-bar');
export const RtkTextComposerView = /*@__PURE__*/ createReactComponent('rtk-text-composer-view');
export const RtkTextMessage = /*@__PURE__*/ createReactComponent('rtk-text-message');
export const RtkTextMessageView = /*@__PURE__*/ createReactComponent('rtk-text-message-view');
export const RtkTooltip = /*@__PURE__*/ createReactComponent('rtk-tooltip');
export const RtkTranscript = /*@__PURE__*/ createReactComponent('rtk-transcript');
export const RtkTranscripts = /*@__PURE__*/ createReactComponent('rtk-transcripts');
export const RtkUiProvider = /*@__PURE__*/ createReactComponent('rtk-ui-provider');
export const RtkViewerCount = /*@__PURE__*/ createReactComponent('rtk-viewer-count');
export const RtkVirtualizedParticipantList = /*@__PURE__*/ createReactComponent('rtk-virtualized-participant-list');
export const RtkWaitingScreen = /*@__PURE__*/ createReactComponent('rtk-waiting-screen');
