// @cloudflare/realtimekit-ui/dist/components/p-3f49a2db.js
import { p as proxyCustomElement, H, h, c as Host } from './p-ae373651.js';
import { e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';

const rtkMenuListCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:flex;width:-moz-max-content;width:max-content;flex-direction:column;padding:var(--rtk-space-0, 0px);color:rgb(var(--rtk-colors-text-1000, 255 255 255));border-width:1px;border-style:solid;--tw-border-opacity:1;border-color:rgba(var(--rtk-colors-background-700, 44 44 44) / var(--tw-border-opacity));--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));overflow:hidden;border-radius:var(--rtk-border-radius-sm, 4px);--tw-shadow:0 10px 15px -3px rgb(0 0 0 / 0.1), 0 4px 6px -4px rgb(0 0 0 / 0.1);--tw-shadow-colored:0 10px 15px -3px var(--tw-shadow-color), 0 4px 6px -4px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000), var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);backdrop-filter:blur(60px)}:host(.primary){--tw-border-opacity:1;border-color:rgba(var(--rtk-colors-brand-600, 13 81 253) / var(--tw-border-opacity));--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-brand-300, 73 124 253) / var(--tw-bg-opacity))}";
const RtkMenuListStyle0 = rtkMenuListCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkMenuList = /*@__PURE__*/ proxyCustomElement(class RtkMenuList extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Variant */
        this.menuVariant = 'secondary';
        /** Language */
        this.t = useLanguage();
    }
    render() {
        return (h(Host, { key: '747bcb38ca155d89fd5ae2c5fa2e667ad1e96a2f', class: { [this.menuVariant]: true } }, h("slot", { key: '5e9137c067847c3c4ef2defdcc958c077b821621' })));
    }
    static get style() { return RtkMenuListStyle0; }
}, [1, "rtk-menu-list", {
        "iconPack": [16],
        "menuVariant": [1, "menu-variant"],
        "t": [16]
    }]);
__decorate([
    SyncWithStore()
], RtkMenuList.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkMenuList.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-menu-list"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-menu-list":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkMenuList);
            }
            break;
    } });
}
defineCustomElement();

export { RtkMenuList as R, defineCustomElement as d };
