// @cloudflare/realtimekit-ui/dist/components/p-61b5e60a.js
import { p as proxyCustomElement, H, h } from './p-ae373651.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';
import { e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import './p-a83ccdbd.js';
import { d as defineCustomElement$3 } from './p-85587a5c.js';
import { d as defineCustomElement$2 } from './p-3e0452c4.js';
import { d as defineCustomElement$1 } from './p-98c3a8f8.js';

const rtkEmojiPickerButtonCss = ":host{}";
const RtkEmojiPickerButtonStyle0 = rtkEmojiPickerButtonCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkEmojiPickerButton = /*@__PURE__*/ proxyCustomElement(class RtkEmojiPickerButton extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        return (h("rtk-tooltip", { key: '82c39530cddfd5cab8672cc5f7f590587749f9cc', label: this.t('chat.send_emoji') }, h("rtk-button", { key: '8e7ee7dbe901b78482ad3f828bb28bda1fdc36b7', variant: "ghost", kind: "icon", class: { active: this.isActive }, title: this.t('chat.send_emoji') }, h("rtk-icon", { key: 'a83691903dae86b3cb7c8cf17fb488e4da13090b', icon: this.iconPack.emoji_multiple }))));
    }
    static get style() { return RtkEmojiPickerButtonStyle0; }
}, [1, "rtk-emoji-picker-button", {
        "iconPack": [16],
        "t": [16],
        "isActive": [4, "is-active"]
    }]);
__decorate([
    SyncWithStore()
], RtkEmojiPickerButton.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkEmojiPickerButton.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-emoji-picker-button", "rtk-button", "rtk-icon", "rtk-tooltip"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-emoji-picker-button":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkEmojiPickerButton);
            }
            break;
        case "rtk-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "rtk-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "rtk-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { RtkEmojiPickerButton as R, defineCustomElement as d };
