// @cloudflare/realtimekit-ui/dist/components/p-63b97c6d.js
import { p as proxyCustomElement, H, e as createEvent, h, c as Host } from './p-ae373651.js';
import { C as ChatHead } from './p-38d22d57.js';
import { a as sanitizeLink } from './p-efabd2ce.js';
import { e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import { d as downloadFile } from './p-47314bc7.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';
import { d as defineCustomElement$3 } from './p-85587a5c.js';
import { d as defineCustomElement$2 } from './p-3e0452c4.js';
import { d as defineCustomElement$1 } from './p-a96efd29.js';

const rtkImageMessageCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}.image-spinner{cursor:wait}.image-errored{cursor:not-allowed}";
const RtkImageMessageStyle0 = rtkImageMessageCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkImageMessage = /*@__PURE__*/ proxyCustomElement(class RtkImageMessage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.stateUpdate = createEvent(this, "rtkStateUpdate", 7);
        /** Date object of now, to calculate distance between dates */
        this.now = new Date();
        /** Whether the message is continued by same user */
        this.isContinued = false;
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        /** show message in bubble */
        this.showBubble = false;
        this.status = 'loading';
    }
    render() {
        return (h(Host, { key: '0fd97958347fefb4a1666e1d676726d13220e5b3' }, !this.isContinued && (h(ChatHead, { key: '8a629fe3782724a0c82c57850ee36ede941d83d7', name: this.message.displayName, time: this.message.time, now: this.now })), h("div", { key: '8e132ad25a04ddb4abb167b6b0e9863670baefac', class: {
                body: true,
                bubble: this.showBubble,
            }, part: "body" }, h("div", { key: '19cf2331068bc84218b2b7153c49ff1af2dedca8', class: { image: true, loaded: this.status === 'loaded' } }, h("img", { key: '86e08153c32cb6f78d8477fc1abdb5b8f4cd0ed0', src: sanitizeLink(this.message.link), onLoad: () => {
                this.status = 'loaded';
            }, onError: () => {
                this.status = 'errored';
            }, onClick: () => {
                if (this.status === 'loaded') {
                    this.stateUpdate.emit({ image: this.message });
                }
            } }), this.status === 'loading' && (h("div", { key: '367b76997445b7df0226aaa063b884c764a6ba63', class: "image-spinner", title: this.t('chat.img.loading'), "aria-label": this.t('chat.img.loading') }, h("rtk-spinner", { key: '704b81940f6f49cb79d2a5976a2714039358e007', iconPack: this.iconPack }))), this.status === 'errored' && (h("div", { key: 'eb9a5b3593051905c3b4b345c810b318da1cb7cf', class: "image-errored", title: this.t('chat.error.img_not_found'), "aria-label": this.t('chat.error.img_not_found') }, h("rtk-icon", { key: '73dfa63e1a253eac3dc75e598341335f63abb855', icon: this.iconPack.image_off }))), this.status === 'loaded' && (h("div", { key: '48444ac8542f8e3b5ed81c93c1b1b1f0b2cac39b', class: "actions" }, h("rtk-button", { key: '2982aed5152442362a22f873cc7070b60462f7e4', class: "action", variant: "secondary", kind: "icon", onClick: () => {
                this.stateUpdate.emit({ image: this.message });
            } }, h("rtk-icon", { key: 'c8dbeac2a347245f21eef52f0c42772e48741f93', icon: this.iconPack.full_screen_maximize })), h("rtk-button", { key: 'cfc3845ea0ae2373bb5a92d897d926797713cbf0', class: "action", variant: "secondary", kind: "icon", onClick: () => downloadFile(this.message.link, { fallbackName: 'image' }) }, h("rtk-icon", { key: '56a6109c8ba88160ef75f8c13923ab12a4764e0e', icon: this.iconPack.download }))))))));
    }
    static get style() { return RtkImageMessageStyle0; }
}, [0, "rtk-image-message", {
        "message": [16],
        "now": [16],
        "isContinued": [516, "is-continued"],
        "iconPack": [16],
        "t": [16],
        "showBubble": [4, "show-bubble"],
        "status": [32]
    }]);
__decorate([
    SyncWithStore()
], RtkImageMessage.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkImageMessage.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-image-message", "rtk-button", "rtk-icon", "rtk-spinner"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-image-message":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkImageMessage);
            }
            break;
        case "rtk-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "rtk-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "rtk-spinner":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { RtkImageMessage as R, defineCustomElement as d };
