// @cloudflare/realtimekit-ui/dist/components/p-758bfd06.js
import { p as proxyCustomElement, H, e as createEvent, h, c as Host } from './p-ae373651.js';
import { e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';
import { d as defineCustomElement$1 } from './p-85587a5c.js';

const rtkMuteAllConfirmationCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{display:block;-webkit-user-select:none;-moz-user-select:none;user-select:none}.leave-modal{position:relative;display:flex;flex-direction:column;border-radius:var(--rtk-border-radius-md, 8px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-900, 26 26 26) / var(--tw-bg-opacity));padding:var(--rtk-space-4, 16px);color:rgb(var(--rtk-colors-text-1000, 255 255 255))}.leave-modal .header h2{margin:var(--rtk-space-0, 0px);margin-bottom:var(--rtk-space-3, 12px)}.leave-modal .content{font-size:14px}.leave-message p{margin-top:var(--rtk-space-3, 12px);margin-bottom:var(--rtk-space-3, 12px)}.leave-meeting{display:flex;flex-direction:row;justify-content:space-between;gap:var(--rtk-space-2, 8px)}.leave-meeting rtk-button{flex:1 1 0%}.end-meeting{margin-top:var(--rtk-space-2, 8px)}.message{margin-bottom:var(--rtk-space-4, 16px)}label{display:flex;align-items:center}input[type='checkbox']{margin-right:var(--rtk-space-2, 8px);accent-color:rgb(var(--rtk-colors-brand-500, 33 96 253))}";
const RtkMuteAllConfirmationStyle0 = rtkMuteAllConfirmationCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkMuteAllConfirmation = /*@__PURE__*/ proxyCustomElement(class RtkMuteAllConfirmation extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.stateUpdate = createEvent(this, "rtkStateUpdate", 7);
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.allowUnmute = true;
        this.onClose = () => {
            this.stateUpdate.emit({ activeMuteAllConfirmation: false });
        };
        this.onMuteAll = () => {
            var _a;
            (_a = this.meeting) === null || _a === void 0 ? void 0 : _a.participants.disableAllAudio(this.allowUnmute);
            this.onClose();
        };
    }
    render() {
        return (h(Host, { key: 'cc42523605a89e127a34ffb366386c96e9e8bcc8' }, h("div", { key: '680059dc6adedc51406964516092716a4be9b0b8', class: "leave-modal" }, h("div", { key: 'c93fd02d2980e40ea05c2a72cde974c13313a5c6', class: "header" }, h("h2", { key: 'ed72b35943df3b6e277b072a283bc14df478334c', class: "title" }, this.t('mute_all.header'))), h("p", { key: '5e0fad3eaa18529f9ae576d0878b81bd8ecd86ec', class: "message" }, this.t('mute_all.description')), h("div", { key: '9d7de3d65427027e8ed91168a2d5005c4786fd03', class: "content" }, h("div", { key: 'aa6ec55ea8d2a57810c7270abeda998439339b0d', class: "leave-meeting" }, h("rtk-button", { key: '98e75f8a7d6f929a988cb79bd4ab0edd55032933', variant: "secondary", title: this.t('close'), onClick: this.onClose }, this.t('cancel')), h("rtk-button", { key: '1e4738bcc21c4f9a09f833028f25b2a9255e718d', variant: "danger", title: this.t('mute_all'), onClick: this.onMuteAll }, this.t('mute_all')))))));
    }
    static get style() { return RtkMuteAllConfirmationStyle0; }
}, [1, "rtk-mute-all-confirmation", {
        "meeting": [16],
        "states": [16],
        "iconPack": [16],
        "t": [16],
        "allowUnmute": [32]
    }]);
__decorate([
    SyncWithStore()
], RtkMuteAllConfirmation.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], RtkMuteAllConfirmation.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], RtkMuteAllConfirmation.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkMuteAllConfirmation.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-mute-all-confirmation", "rtk-button"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-mute-all-confirmation":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkMuteAllConfirmation);
            }
            break;
        case "rtk-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { RtkMuteAllConfirmation as R, defineCustomElement as d };
