// @cloudflare/realtimekit-ui/dist/components/p-b5ee1586.js
import { p as proxyCustomElement, H, h, c as Host } from './p-ae373651.js';
import { C as ChatHead } from './p-38d22d57.js';
import { a as sanitizeLink } from './p-efabd2ce.js';
import { e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import { g as getExtension, a as getFileSize, d as downloadFile } from './p-47314bc7.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';
import { d as defineCustomElement$2 } from './p-85587a5c.js';
import { d as defineCustomElement$1 } from './p-3e0452c4.js';

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkFileMessage = /*@__PURE__*/ proxyCustomElement(class RtkFileMessage extends H {
    constructor() {
        super();
        this.__registerHost();
        /** Date object of now, to calculate distance between dates */
        this.now = new Date();
        /** Whether the message is continued by same user */
        this.isContinued = false;
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        /** show message in bubble */
        this.showBubble = false;
    }
    render() {
        const link = sanitizeLink(this.message.link);
        return (h(Host, { key: '76c8da9d878e7c1fb3b8ffa315237cd913338276' }, !this.isContinued && (h(ChatHead, { key: 'efd7339f8335d7da577f61caa61642fe48f5a412', name: this.message.displayName, time: this.message.time, now: this.now })), h("div", { key: '53947f39384592b44f42199a3c8eb2f4d90adafd', class: {
                body: true,
                bubble: this.showBubble,
            }, part: "body" }, h("div", { key: '1a09d5b047cfa88912e8fada39a5cf8b0e138736', class: "file" }, h("div", { key: 'fc6a30c8ddb0f8ac895de7ffad4bc4c455a59dfa', class: "file-data" }, h("div", { key: '4ad8fa217942ca3f61faea3ed4de9fd0f158bbb1', class: "name" }, this.message.name), h("div", { key: 'd4adc2dd11b6be4ce508ac50eea67e8d500f1929', class: "file-data-split" }, h("div", { key: 'cca022a6825fbafa4c257ac9f32f4d25f2eece2b', class: "ext" }, getExtension(this.message.name)), h("span", { key: '62917b4cb5371045f8fcee541c2cc872038a90f9', class: "divider" }), h("div", { key: '9cae7724db8096dfa64f2f2e70915fddd0b1fb43', class: "size" }, getFileSize(this.message.size)))), h("rtk-button", { key: 'd2645e34ed2f984fa0403820c47329fa82ae4d58', variant: "secondary", kind: "icon", onClick: () => downloadFile(link, { name: this.message.name, fallbackName: 'file' }), part: "button" }, h("rtk-icon", { key: '5650ea4473efec76a2e8942fa32be671ac143fb7', icon: this.iconPack.download }))))));
    }
}, [0, "rtk-file-message", {
        "message": [16],
        "now": [16],
        "isContinued": [516, "is-continued"],
        "iconPack": [16],
        "t": [16],
        "showBubble": [4, "show-bubble"]
    }]);
__decorate([
    SyncWithStore()
], RtkFileMessage.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkFileMessage.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-file-message", "rtk-button", "rtk-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-file-message":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkFileMessage);
            }
            break;
        case "rtk-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "rtk-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { RtkFileMessage as R, defineCustomElement as d };
