// @cloudflare/realtimekit-ui/dist/components/rtk-controlbar.js
import { p as proxyCustomElement, H, h, c as Host } from './p-ae373651.js';
import { c as createDefaultConfig, e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import './p-a83ccdbd.js';
import { R as Render } from './p-88bd1053.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';

const rtkControlbarCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{box-sizing:border-box;display:flex;align-items:center;gap:var(--rtk-space-0\\.5, 2px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--rtk-space-4, 16px);padding-right:var(--rtk-space-4, 16px);position:relative;z-index:10}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{padding-top:var(--rtk-space-0, 0px) !important}}";
const RtkControlbarStyle0 = rtkControlbarCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkControlbar$1 = /*@__PURE__*/ proxyCustomElement(class RtkControlbar extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = createDefaultConfig();
        /** Icon Pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (h(Host, { key: 'be00ae26f16ab4381eddade1f80172789a2833b2' }, !this.disableRender && (h(Render, { key: 'd7ed714d51dfcb94195864d462c1b9598eea8ca7', element: "rtk-controlbar", defaults: defaults, onlyChildren: true })), h("slot", { key: '1601a47d06eab5c79762abdb45d3edb4b0a9f6b6' })));
    }
    static get style() { return RtkControlbarStyle0; }
}, [1, "rtk-controlbar", {
        "variant": [513],
        "disableRender": [4, "disable-render"],
        "meeting": [16],
        "config": [16],
        "states": [16],
        "iconPack": [16],
        "t": [16],
        "size": [513]
    }]);
__decorate([
    SyncWithStore()
], RtkControlbar$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], RtkControlbar$1.prototype, "config", void 0);
__decorate([
    SyncWithStore()
], RtkControlbar$1.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], RtkControlbar$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkControlbar$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-controlbar"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-controlbar":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkControlbar$1);
            }
            break;
    } });
}
defineCustomElement$1();

const RtkControlbar = RtkControlbar$1;
const defineCustomElement = defineCustomElement$1;

export { RtkControlbar, defineCustomElement };
