// @cloudflare/realtimekit-ui/dist/components/rtk-idle-screen.js
import { p as proxyCustomElement, H, h, c as Host } from './p-ae373651.js';
import { c as createDefaultConfig, e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import './p-a83ccdbd.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';
import { d as defineCustomElement$4 } from './p-3e0452c4.js';
import { d as defineCustomElement$3 } from './p-c270aaf5.js';
import { d as defineCustomElement$2 } from './p-a96efd29.js';

const rtkIdleScreenCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{height:100%;width:100%;display:flex;flex-direction:column;align-items:center;justify-content:center}.ctr{display:flex;flex-direction:column;align-items:center;gap:var(--rtk-space-8, 32px)}.no-network-badge{margin-top:var(--rtk-space-2, 8px);display:flex;width:100%;flex-direction:row;align-items:center;justify-content:flex-start;border-radius:var(--rtk-border-radius-sm, 4px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-brand-500, 33 96 253) / var(--tw-bg-opacity));background-color:rgba(var(--rtk-colors-danger, 255 45 45) / 0.1);padding-top:var(--rtk-space-1, 4px);padding-bottom:var(--rtk-space-1, 4px);font-size:12px;color:rgba(var(--rtk-colors-danger, 255 45 45) / 0.75)}.no-network-badge rtk-icon{margin-left:var(--rtk-space-2, 8px);margin-right:var(--rtk-space-2, 8px)}rtk-logo.loaded{height:var(--rtk-space-12, 48px)}rtk-spinner{height:var(--rtk-space-12, 48px);width:var(--rtk-space-12, 48px);--tw-text-opacity:1;color:rgba(var(--rtk-colors-brand-500, 33 96 253) / var(--tw-text-opacity))}";
const RtkIdleScreenStyle0 = rtkIdleScreenCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkIdleScreen$1 = /*@__PURE__*/ proxyCustomElement(class RtkIdleScreen extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Config object */
        this.config = createDefaultConfig();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.socketStateUpdate = ({ state }) => {
            var _a;
            this.connectionState = state;
            if (state === 'connected') {
                if (!((_a = this.states) === null || _a === void 0 ? void 0 : _a.joinError)) {
                    this.joinError = undefined;
                }
            }
        };
    }
    connectedCallback() {
        this.meetingChanged(this.meeting, undefined);
        this.statesChanged(this.states);
    }
    disconnectedCallback() {
        var _a, _b;
        (_b = (_a = this.meeting) === null || _a === void 0 ? void 0 : _a.meta) === null || _b === void 0 ? void 0 : _b.removeListener('socketConnectionUpdate', this.socketStateUpdate);
    }
    meetingChanged(meeting, oldMeeting) {
        var _a, _b, _c, _d;
        (_a = oldMeeting === null || oldMeeting === void 0 ? void 0 : oldMeeting.meta) === null || _a === void 0 ? void 0 : _a.removeListener('socketConnectionUpdate', this.socketStateUpdate);
        if (!meeting)
            return;
        this.connectionState = (_c = (_b = meeting.meta) === null || _b === void 0 ? void 0 : _b.socketState) === null || _c === void 0 ? void 0 : _c.state;
        (_d = meeting.meta) === null || _d === void 0 ? void 0 : _d.addListener('socketConnectionUpdate', this.socketStateUpdate);
    }
    statesChanged(states) {
        this.joinError = states === null || states === void 0 ? void 0 : states.joinError;
    }
    render() {
        const showSocketError = !!this.connectionState && this.connectionState !== 'connected' && !this.joinError;
        const errorText = this.joinError
            ? this.joinError
            : this.connectionState === 'failed'
                ? this.t('network.lost_extended')
                : this.t('network.lost');
        return (h(Host, { key: '5b33a77fbe2d968dc0c54d7c6ff3894202df0d94' }, h("slot", { key: 'eac40b23ef0198400c376606a1e5ff575c4b9014' }, h("div", { key: '121ffb27b1e9c35dd8b15f0eb6a25c00226416ac', class: "ctr", part: "container" }, h("rtk-logo", { key: 'eb2230d7dc1915df6469427767a9beab5dd63eb3', meeting: this.meeting, config: this.config, t: this.t, part: "logo" }), this.joinError || showSocketError ? (h("div", { class: "no-network-badge", part: "network-badge" }, h("rtk-icon", { size: "md", variant: "danger", icon: this.iconPack.disconnected, part: "network-badge-icon" }), errorText)) : (h("rtk-spinner", { "aria-label": "Idle, waiting for meeting data", part: "spinner", iconPack: this.iconPack }))))));
    }
    static get watchers() { return {
        "meeting": ["meetingChanged"],
        "states": ["statesChanged"]
    }; }
    static get style() { return RtkIdleScreenStyle0; }
}, [1, "rtk-idle-screen", {
        "meeting": [16],
        "states": [16],
        "config": [16],
        "iconPack": [16],
        "t": [16],
        "joinError": [32],
        "connectionState": [32]
    }, undefined, {
        "meeting": ["meetingChanged"],
        "states": ["statesChanged"]
    }]);
__decorate([
    SyncWithStore()
], RtkIdleScreen$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], RtkIdleScreen$1.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], RtkIdleScreen$1.prototype, "config", void 0);
__decorate([
    SyncWithStore()
], RtkIdleScreen$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkIdleScreen$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-idle-screen", "rtk-icon", "rtk-logo", "rtk-spinner"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-idle-screen":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkIdleScreen$1);
            }
            break;
        case "rtk-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "rtk-logo":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "rtk-spinner":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const RtkIdleScreen = RtkIdleScreen$1;
const defineCustomElement = defineCustomElement$1;

export { RtkIdleScreen, defineCustomElement };
