// @cloudflare/realtimekit-ui/dist/components/rtk-waiting-screen.js
import { p as proxyCustomElement, H, h, c as Host } from './p-ae373651.js';
import { c as createDefaultConfig, e as defaultIconPack, i as useLanguage } from './p-81fcdba3.js';
import './p-a83ccdbd.js';
import { S as SyncWithStore } from './p-f8a8c06f.js';
import { d as defineCustomElement$2 } from './p-c270aaf5.js';

const rtkWaitingScreenCss = ":host{line-height:initial;font-family:var(--rtk-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--rtk-space-0, 0px);padding:var(--rtk-space-0, 0px)}:host{height:100%;width:100%;display:flex;flex-direction:column;align-items:center;justify-content:center}.centered{display:flex;flex-direction:column;align-items:center}rtk-logo{margin-bottom:var(--rtk-space-8, 32px);height:var(--rtk-space-12, 48px)}p{font-size:16px;border-radius:var(--rtk-border-radius-lg, 12px);--tw-bg-opacity:1;background-color:rgba(var(--rtk-colors-background-800, 30 30 30) / var(--tw-bg-opacity));padding-left:var(--rtk-space-8, 32px);padding-right:var(--rtk-space-8, 32px);padding-top:var(--rtk-space-4, 16px);padding-bottom:var(--rtk-space-4, 16px);color:rgb(var(--rtk-colors-text-1000, 255 255 255))}";
const RtkWaitingScreenStyle0 = rtkWaitingScreenCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const RtkWaitingScreen$1 = /*@__PURE__*/ proxyCustomElement(class RtkWaitingScreen extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Config */
        this.config = createDefaultConfig();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        return (h(Host, { key: 'ac7096bd5df59387dcef15bbd4ef8955a4a42298' }, h("slot", { key: '6dd0f5b9555c956c1371311ef3c4ac8f661e9523' }, h("div", { key: '07925b636545e1196b10512cd504516df2cfbbf4', class: "centered", part: "content" }, h("rtk-logo", { key: '9b3fcf182d7c68aaf525a8170dbab9e59fc40404', meeting: this.meeting, config: this.config, part: "logo", t: this.t }), h("p", { key: '763a6403bc98a16f020101090bbed250041ae7cd' }, this.t('waitlist.body_text'))))));
    }
    static get style() { return RtkWaitingScreenStyle0; }
}, [1, "rtk-waiting-screen", {
        "meeting": [16],
        "config": [16],
        "iconPack": [16],
        "t": [16]
    }]);
__decorate([
    SyncWithStore()
], RtkWaitingScreen$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], RtkWaitingScreen$1.prototype, "config", void 0);
__decorate([
    SyncWithStore()
], RtkWaitingScreen$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], RtkWaitingScreen$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["rtk-waiting-screen", "rtk-logo"];
    components.forEach(tagName => { switch (tagName) {
        case "rtk-waiting-screen":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, RtkWaitingScreen$1);
            }
            break;
        case "rtk-logo":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const RtkWaitingScreen = RtkWaitingScreen$1;
const defineCustomElement = defineCustomElement$1;

export { RtkWaitingScreen, defineCustomElement };
