// @cloudflare/realtimekit-ui/dist/collection/components/rtk-chat-header/rtk-chat-header.js
import { h, Host } from "@stencil/core";
export class RtkChatHeader {
    constructor() {
        this.onPinnedToggle = async (e) => {
            var _a, _b, _c;
            if ((_a = e.detail) === null || _a === void 0 ? void 0 : _a.open) {
                await ((_c = (_b = this.$chatSelector) === null || _b === void 0 ? void 0 : _b.close) === null || _c === void 0 ? void 0 : _c.call(_b));
            }
        };
        this.onChatToggle = async (e) => {
            var _a, _b, _c;
            if ((_a = e.detail) === null || _a === void 0 ? void 0 : _a.open) {
                await ((_c = (_b = this.$pinnedSelector) === null || _b === void 0 ? void 0 : _b.close) === null || _c === void 0 ? void 0 : _c.call(_b));
            }
        };
    }
    render() {
        return (h(Host, { key: 'af9956ff6e693f29fb63a6180ef773a0e64822e5' }, h("rtk-pinned-message-selector", { key: 'b0d9199d02d84453f8ee26c3a47e00c82047e4f6', ref: (el) => (this.$pinnedSelector = el), onRtkDropdownToggle: this.onPinnedToggle }), h("rtk-chat-selector", { key: '4d019562a85ebfeaf4ceae2e99d1b9c8c282127a', ref: (el) => (this.$chatSelector = el), onRtkDropdownToggle: this.onChatToggle })));
    }
    static get is() { return "rtk-chat-header"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["rtk-chat-header.css"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["rtk-chat-header.css"]
        };
    }
}
